### PR TITLE
ec2_vol: Add key selection support for EBS volume encryption.

### PR DESCRIPTION
**Pull request by bachp**
Original-code-by: https://github.com/nonshankus
More or less a rebase of PR onto master: https://github.com/ansible/ansible-modules-core/issues/3217

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
ec2_vol

##### ANSIBLE VERSION
```
ansible 2.1.1.0
```

##### SUMMARY
Add support to specify the encryption key for encrypted EBS volumes.

It adds an additional config `kms_key_id` that can be used to use a key other than the default.

### CODE REVIEW

File: lib/ansible/modules/cloud/amazon/ec2_vol.py
Comment (ryansb) [inline lib/ansible/modules/cloud/amazon/ec2_vol.py]: This check should be `kms_key_id is not None`

File: lib/ansible/modules/cloud/amazon/ec2_vol.py
Comment (ryansb) [inline lib/ansible/modules/cloud/amazon/ec2_vol.py]: This check should be `kms_key_id is not None`

**Review by ryansb — CHANGES_REQUESTED** [review]

**Comment by willthames** [comment]
:-1: I see this as misappropriation of a commit.

It's not hard to rebase a commit while retaining original authorship*. Given that further rework is apparently required on top, you can add your own commit to fix things.

\* git-format-patch + git-am is one approach with which I've had a great deal of success.

**Comment by bachp** [comment]
@willthames Could you point me to some instructions on how to do the rebase? The problem I was facing is that the location of the files changed when they were moved to the ansible repository.

**Comment by willthames** [comment]
@bachp sure, no problem, here we go:

```
cd path/to/ansible-modules-core
git fetch upstream pull/3218/head:pr-3218
git checkout pr-3218
git format-patch upstream/devel -o /tmp
```

Then to apply that same patch

```
cd path/to/ansible
git checkout -b ec2_enc_keys devel
git am -p1 --directory lib/ansible/modules /tmp/000*
```

**Comment by bachp** [comment]
@ryansb I addressed your issues.

@willthames Thanks the `--directory` option was the one I was not aware of. Will sure be helpful in the future again.

**Comment by willthames** [comment]
ready_for_review

**Comment by nbasinge** [comment]
What is the status? Thanks.

**Comment by ryansb** [comment]
LGTM, just needs a maintainer approval and/or community shipits.

**Review by ryansb — APPROVED** [review]

**Comment by willthames** [comment]
shipit

**Comment by willthames** [comment]
@ryansb not sure community review has been requested here - ansibot hasn't cc'd anyone for review.

**Comment by willthames** [comment]
!needs_revision

**Review by ryansb — APPROVED** [review]

**Pull request merged**
